arelle/CntlrCmdLine.py

```diff
@@ -1083,6 +1090,16 @@ def monitorParentProcess():
                             ViewFileRoleTypes.viewRoleTypes(modelXbrl, options.roleTypesFile, "Role Types", isArcrole=False, lang=options.labelLang)
                         if options.arcroleTypesFile:
                             ViewFileRoleTypes.viewRoleTypes(modelXbrl, options.arcroleTypesFile, "Arcrole Types", isArcrole=True, lang=options.labelLang)
+                        if options.saveDeduplicatedInstance:
+                            if options.deduplicateFacts:
+                                deduplicateFactsArg = ValidateDuplicateFacts.DeduplicationType(options.deduplicateFacts)
```

Comment:
Does this need an assertion (and documentation) that this should only be used with traditional XBRL?

What happens if this runs on an inline doc? Does it remove the fact and value from the HTML? I guess a stretch goal could be to remove the fact tagging from iXBRL while preserving the rendered number (replace the XBRL element with its inner text), but I think exiting with an error is good enough until someone asks us to support iXBRL. iXBRL support has limited (no?) value while introducing many complexities (document sets, multiple targets, etc.)

Also, I think this should probably exit after saving the deduplicated instance instead of continuing on (log an error if the user tries to do other actions, validate, etc.) It appears that this puts the model into a strange state (we've modified the lxml tree to remove facts, but the model XBRL still includes them all).

Reply:
Opted to just raise an error if this is run on any document that isn't a traditional XBRL instance. I do think iXBRL deduplication would be possible, but it makes sense to wait for demand.

I moved deduplication to run as late as possible since it does put the model into an undefined state. For what it's worth, I do think there's a way to write a new deduplicated XBRL document without manipulating the in-memory ModelXbrl (probably involves walking the lxml tree and writing every element that *isn't* a duplicate fact), but doing it this way allowed reuse of `ModelXbrl.saveInstance`. I'm willing to explore this approach if there's a consensus that it is worthwhile.

It seems that the only case where the model would potentially be used after the model is manipulated is if `keepOpen` is active or plugins that make use of `CntlrCmdLine.Filing.Validate` or `CntlrCmdLine.Filing.End` are active.

Reply:
`CntlrCmdLine.Filing.End` and `CntlrCmdLine.Filing.Validate` are pretty common hooks. Can you move this after those hooks and raise an exception if `keepOpen` is used along with `saveDeduplicatedInstance`?

Reply:
I've updated this to raise an error after saving if `keepOpen` is used.

The `ModelXbrl` is closed immediately after deduplication occurs currently (unless `keepOpen` is enabled), so it can't be moved any later. Looking into it more, it would appear that if either `CntlrCmdLine.Filing.End` and `CntlrCmdLine.Filing.Validate` are used by a plugin, they won't have access to an open `ModelXbrl` unless they're also using `keepOpen`, so I don't think moving this is necessary.

Reply:
Thanks. I missed that the models were closed before those hooks executed.